naga/src/arena.rs

```diff
@@ -297,6 +297,15 @@ impl<T> Arena<T> {
             .map(|(i, v)| unsafe { (Handle::from_usize_unchecked(i), v) })
     }
 
+    /// Drains the arena, returning an iterator over the items stored.
+    pub fn drain(self) -> impl DoubleEndedIterator<Item = (Handle<T>, T, Span)> {
+        self.data
+            .into_iter()
+            .zip(self.span_info.into_iter())
+            .enumerate()
+            .map(|(i, (v, span))| unsafe { (Handle::from_usize_unchecked(i), v, span) })
+    }
```

Comment:
This actually works - with corresponding changes to `process_overrides`. The return value doesn't seem to extend the lifetime of the `&mut self`, so `process_overrides` is free to mutate `module`.

```suggestion
    pub fn drain(&mut self) -> impl DoubleEndedIterator<Item = (Handle<T>, T, Span)> {
        let arena = std::mem::take(self);
        arena.data
            .into_iter()
            .zip(arena.span_info.into_iter())
            .enumerate()
            .map(|(i, (v, span))| unsafe { (Handle::from_usize_unchecked(i), v, span) })
    }
```